Failed to build L2CAPContext.java  and Peripheral.java

Hello.

I'm trying to build an ionic project that uses this plugin and 4 months earlier I was able to build it but no more.

More specifically here is what I'm getting: 

```
/home/user/dev/ionic-app/platforms/android/app/src/main/java/com/megster/cordova/ble/central/Peripheral.java:35: error: cannot find symbol
import androidx.annotation.RequiresPermission;
                          ^
  symbol:   class RequiresPermission
  location: package androidx.annotation
/home/user/dev/ionic-app/platforms/android/app/src/main/java/com/megster/cordova/ble/central/L2CAPContext.java:6: error: cannot find symbol
import androidx.annotation.RequiresApi;
                          ^
  symbol:   class RequiresApi
  location: package androidx.annotation
/home/user/dev/ionic-app/platforms/android/app/src/main/java/com/megster/cordova/ble/central/Peripheral.java:1000: error: cannot find symbol
    @RequiresPermission("android.permission.BLUETOOTH_CONNECT")
     ^
  symbol:   class RequiresPermission
  location: class Peripheral
/home/user/dev/ionic-app/platforms/android/app/src/main/java/com/megster/cordova/ble/central/Peripheral.java:1028: error: cannot find symbol
    @RequiresPermission("android.permission.BLUETOOTH_CONNECT")
     ^
  symbol:   class RequiresPermission
  location: class Peripheral
/home/user/dev/ionic-app/platforms/android/app/src/main/java/com/megster/cordova/ble/central/Peripheral.java:1076: error: cannot find symbol
    @RequiresPermission("android.permission.BLUETOOTH_CONNECT")
     ^
  symbol:   class RequiresPermission
  location: class Peripheral
/home/user/dev/ionic-app/platforms/android/app/src/main/java/com/megster/cordova/ble/central/L2CAPContext.java:110: error: cannot find symbol
    @RequiresApi(api = Build.VERSION_CODES.M)
     ^
  symbol:   class RequiresApi
  location: class L2CAPContext
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
6 errors

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:compileReleaseJavaWithJavac'.
```

The environment of mine is quite old but I'm not sure if that's the issue. 

```
$ npm list -g

/home/user/.nvm/versions/node/v16.9.0/lib
├── @angular/cli@8.3.29
├── @ionic/cli@6.17.1
├── cordova@10.0.0
├── ionic@5.4.16
└── npm@7.21.1

$ java -version
java version "1.8.0_301"
```

In Ionic's config file I have set `<preference name="android-targetSdkVersion" value="31" />`

Plugin's version: 1.7.0

--- a/.github/workflows/capacitor.yaml
+++ b/.github/workflows/capacitor.yaml
@@ -22,9 +22,6 @@ jobs:
           - jdk: 11
             capacitor: 4
             node: 14.x
-          - jdk: 8
-            capacitor: 3
-            node: 14.x
 
     steps:
       - uses: actions/checkout@v3

--- a/.github/workflows/capacitor.yaml
+++ b/.github/workflows/capacitor.yaml
@@ -3,16 +3,16 @@ name: Capacitor
 on:
   workflow_dispatch:
   push:
-    branches: [ "master" ]
+    branches: ['master']
   pull_request:
-    branches: [ "master" ]
+    branches: ['master']
 
 jobs:
   capacitor-android:
     name: android@${{ matrix.capacitor }}
-    
+
     runs-on: ubuntu-latest
-    
+
     strategy:
       matrix:
         include:
@@ -24,37 +24,36 @@ jobs:
             node: 14.x
 
     steps:
-    - uses: actions/checkout@v3
-    - name: set up JDK ${{ matrix.jdk }}
-      uses: actions/setup-java@v3
-      with:
-        java-version: ${{ matrix.jdk }}
-        distribution: 'temurin'
-    - name: Use Node.js ${{ matrix.node }}
-      uses: actions/setup-node@v3
-      with:
-        node-version: ${{ matrix.node }}
-    - name: Build test app
-      run: |
-        mkdir temp
-        cd temp
-        mkdir www
-        touch www/index.html
-        npm init -y
-        npm install ..
-        npm install @capacitor/cli@${{ matrix.capacitor }} @capacitor/core@${{ matrix.capacitor }} @capacitor/android@${{ matrix.capacitor }}
-        npx cap init test io.test.app
-        npx cap add android
-        npx cap sync android
-        cd android
-        chmod +x gradlew
-        ./gradlew build
-        
+      - uses: actions/checkout@v3
+      - name: set up JDK ${{ matrix.jdk }}
+        uses: actions/setup-java@v3
+        with:
+          java-version: ${{ matrix.jdk }}
+          distribution: 'temurin'
+      - name: Use Node.js ${{ matrix.node }}
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node }}
+      - name: Build test app
+        run: |
+          mkdir temp
+          cd temp
+          mkdir www
+          touch www/index.html
+          npm init -y
+          npm install ..
+          npm install @capacitor/cli@${{ matrix.capacitor }} @capacitor/core@${{ matrix.capacitor }} @capacitor/android@${{ matrix.capacitor }}
+          npx cap init test io.test.app
+          npx cap add android
+          npx cap sync android
+          cd android
+          chmod +x gradlew
+          ./gradlew build
 
   capacitor-ios:
     name: ios@${{ matrix.capacitor }}
     runs-on: macos-latest
-    
+
     strategy:
       matrix:
         include:
@@ -62,22 +61,22 @@ jobs:
             node: 18.x
 
     steps:
-    - uses: actions/checkout@v3
-    - name: Use Node.js ${{ matrix.node }}
-      uses: actions/setup-node@v3
-      with:
-        node-version: ${{ matrix.node }}
-    - name: Build test app
-      run: |
-        mkdir temp
-        cd temp
-        mkdir www
-        touch www/index.html
-        npm init -y
-        npm install ..
-        npm install @capacitor/cli@${{ matrix.capacitor }} @capacitor/core@${{ matrix.capacitor }} @capacitor/ios@${{ matrix.capacitor }}
-        npx cap init test io.test.app
-        npx cap add ios
-        npx cap sync ios
-        cd ios
-        xcodebuild -workspace App/App.xcworkspace -scheme CapacitorCordova -configuration Debug -sdk iphonesimulator -destination "platform=iOS Simulator,name=iPhone 14 Pro Max"
+      - uses: actions/checkout@v3
+      - name: Use Node.js ${{ matrix.node }}
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node }}
+      - name: Build test app
+        run: |
+          mkdir temp
+          cd temp
+          mkdir www
+          touch www/index.html
+          npm init -y
+          npm install ..
+          npm install @capacitor/cli@${{ matrix.capacitor }} @capacitor/core@${{ matrix.capacitor }} @capacitor/ios@${{ matrix.capacitor }}
+          npx cap init test io.test.app
+          npx cap add ios
+          npx cap sync ios
+          cd ios
+          xcodebuild -workspace App/App.xcworkspace -scheme CapacitorCordova -configuration Debug -sdk iphonesimulator -destination "platform=iOS Simulator,name=iPhone 14 Pro Max"

--- a/.github/workflows/capacitor.yaml
+++ b/.github/workflows/capacitor.yaml
@@ -24,7 +24,7 @@ jobs:
             node: 14.x
           - jdk: 8
             capacitor: 3
-            node: 12.x
+            node: 14.x
 
     steps:
       - uses: actions/checkout@v3

--- a/.github/workflows/capacitor.yaml
+++ b/.github/workflows/capacitor.yaml
@@ -22,6 +22,9 @@ jobs:
           - jdk: 11
             capacitor: 4
             node: 14.x
+          - jdk: 8
+            capacitor: 3
+            node: 12.x
 
     steps:
       - uses: actions/checkout@v3

--- a/.prettierrc
+++ b/.prettierrc
@@ -15,6 +15,12 @@
             "options": {
                 "tabWidth": 2
             }
+        },
+        {
+            "files": "*.yaml",
+            "options": {
+                "tabWidth": 2
+            }
         }
     ]
 }

--- a/plugin.xml
+++ b/plugin.xml
@@ -83,6 +83,7 @@
         <source-file src="src/android/Helper.java" target-dir="src/com/megster/cordova/ble/central"/>
         <source-file src="src/android/UUIDHelper.java" target-dir="src/com/megster/cordova/ble/central"/>
         <source-file src="src/android/L2CAPContext.java" target-dir="src/com/megster/cordova/ble/central"/>
+        <framework src="androidx.annotation:annotation:*"/>
     </platform>
 
     <platform name="browser">

